packages/babel-plugin-transform-classes/src/transformClass.js

```diff
@@ -428,7 +410,44 @@ export default function transformClass(
       if (processMethod(node, scope)) return;
     }
 
-    pushToMap(node, false, null, scope);
+    const placement = node.static ? "static" : "instance";
+    const methods = classState.methods[placement];
+
+    const { computed } = node;
+    if (computed) methods.hasComputed = true;
+
+    const key = t.toComputedKey(node);
+    const descKey = node.kind === "method" ? "value" : node.kind;
+
+    let fn = t.toExpression(node);
+    // infer function name
+    if (t.isStringLiteral(key) && node.kind === "method") {
+      fn = nameFunction({ id: key, node: node, scope });
+    }
+
+    let descriptor;
+    if (
+      !methods.hasComputed &&
+      t.isStringLiteral(key) &&
```

Comment:
Not related to this PR, but we should also merge descriptors for NumericLiteral.

```js
class C {
  set 0n(v) {}
  get 0n() {}
  set 1(v) {}
  get 1() {}
}
```

Reply:
This code is new, so I can handle it here (they are merged with `main`).